run-tests.sh

```diff
@@ -6,12 +6,18 @@
 # REANA is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
+RJC_IMAGE_NAME=reanahub/reana-job-controller
+docker build -t $RJC_IMAGE_NAME .
```

Comment:
As mused together IRL:

(1) Please do this only on macOS platform (i.e. docker first and tests inside it); for GNU/Linux let us keep the old order (tests first and docker afterwards), so that the lengthy Docker test is run the last.

(2) Please introduce `./run-tests.sh --include-docker-tests` option that would not be plugged by default (in local testing on laptops where we do have running docker builds naturally already anyway) and that we would be only plugged on for Travis CI tests.

We can simply use the same if-option-not-specified-then-stop technique as in CERN Open Data: https://github.com/cernopendata/opendata.cern.ch/blob/master/run-tests.sh#L76-L79

Keeping exit code 0 even if we don't test docker will help in chaining commands together.

(3) Let's amend all the components that do use docker tests in `run-tests.sh` in the same way. (`r-w-controller`, `r-w-e-*`, `r-server`, `r-w-monitor`... and perhaps `r-ui` too?!)